modules/vm/templates/entrypoint.sh.tpl

```diff
@@ -14,9 +14,102 @@ done
+existingUnattachedDisk=$(az disk list --resource-group $resourceGroup --query "[?diskState=='Unattached' && starts_with(name, 'Disk_$${vmssName}')].{Name:name}" --output tsv)
+
+if [ -z "$existingUnattachedDisk" ]; then
+  echo "Creating a new managed disk"
+  az disk create --resource-group $resourceGroup --name $diskName --size-gb $diskSizeGB --location $regionID --sku Premium_LRS --zone $zoneID --tier $tier
+fi
+
+# Checks if a managed disk is attached to the instance
+attachedDisk=$(az vmss list-instances --resource-group "$resourceGroup" --name "$vmssName" --query "[?instanceId==\"$instanceID\"].storageProfile.dataDisks[].name" --output tsv)
+
+if [ -z "$attachedDisk" ]; then
+    echo "No data disks attached for instance ID $instanceID in VMSS $vmssName."
+    # Try to attach an existing managed disk
+    availableDisks=$(az disk list --resource-group $resourceGroup --query "[?diskState=='Unattached' && starts_with(name, 'Disk_$${vmssName}') && zones[0]=='$${zoneID}'].{Name:name}" --output tsv)
+      echo "Attaching available disk $availableDisks."
+      # Set Internal Field Separator to newline to handle spaces in names
+      IFS=$'\n'
+      # Would iterate through all available disks and attempt to attach them
+      for availableDisk in $availableDisks; do
+        az vmss disk attach --vmss-name $vmssName --resource-group $resourceGroup --instance-id $instanceID --lun $lun --disk "$availableDisk" || true
+      done
+fi
+
+# Gets device name based on LUN
+graphdb_device=$(lsscsi --scsi --size | awk '/\[1:.*:0:2\]/ {print $7}')
+
+# Check if the device is present after attaching the disk
+if [ -b "$graphdb_device" ]; then
+    echo "Device $graphdb_device is available."
+else
+    echo "Device $graphdb_device is not available. Something went wrong."
+    exit 1
+fi
+
+# create a file system if there isn't any
+if [ "$graphdb_device: data" = "$(file -s $graphdb_device)" ]; then
+  mkfs -t ext4 $graphdb_device
+fi
+
+disk_mount_point="/var/opt/graphdb"
+
+# Check if the disk is already mounted
+if ! mount | grep -q "$graphdb_device"; then
+  echo "The disk at $graphdb_device is not mounted."
+
+  # Create the mount point if it doesn't exist
+  if [ ! -d "$disk_mount_point" ]; then
+    mkdir -p "$disk_mount_point"
+  fi
+
+  # Add an entry to the fstab file to automatically mount the disk
+  if ! grep -q "$graphdb_device" /etc/fstab; then
+    echo "$graphdb_device $disk_mount_point ext4 defaults 0 2" >> /etc/fstab
+  fi
+
+  # Mount the disk
+  mount "$disk_mount_point"
+  echo "The disk at $graphdb_device is now mounted at $disk_mount_point."
+else
+  echo "The disk at $graphdb_device is already mounted."
+fi
+
+# Recreates folders if necessary and changes owner
+
+if [ ! -d /var/opt/graphdb/node ]; then
+    mkdir -p /var/opt/graphdb/node
+fi
+
+if [ ! -d /var/opt/graphdb/cluster-proxy ]; then
+    mkdir -p /var/opt/graphdb/cluster-proxy
+fi
+
+chown -R graphdb:graphdb /var/opt/graphdb
```

Comment:
let's add a todo here to research how to fix the permissions and avoid doing this because when there are a lot of files, this operation gets slower.